code/weapon/beam.cpp

```diff
@@ -427,7 +427,7 @@ int beam_fire(beam_fire_info *fire_info)
 	}	
 
 	// create the associated object
-	objnum = obj_create(OBJ_BEAM, ((fire_info->shooter != NULL) ? OBJ_INDEX(fire_info->shooter) : -1), new_item - Beams, &vmd_identity_matrix, &vmd_zero_vector, 1.0f, OF_COLLIDES);
+	objnum = obj_create(OBJ_BEAM, ((fire_info->shooter != NULL) ? OBJ_INDEX(fire_info->shooter) : -1), (int)(new_item - Beams), &vmd_identity_matrix, &vmd_zero_vector, 1.0f, OF_COLLIDES);
```

Comment:
Would it be better to create a new `BEAM_INDEX` macro to be used here instead of `(int)(new_item - Beams)`?